src/Time/Formatting.hs

```diff
@@ -38,10 +50,27 @@ module Time.Formatting
 
 import Time.Rational (Rat)
 #if ( __GLASGOW_HASKELL__ >= 804 )
-import Time.Rational (withRuntimeDivRat)
+import Time.Rational (type (>=%), withRuntimeDivRat)
 #endif
 import Time.Units (AllTimes, KnownRatName, Time, floorUnit, toUnit)
 
+#if ( __GLASGOW_HASKELL__ >= 804 )
+-- Type-level 'if' for 'Bool's.
+type family If (c :: Bool) (t :: Bool) (e :: Bool) where
+  If 'True  t e = t
+  If 'False t e = e
```

Comment:
I think the code can become a little bit nicer if instead of `If` you write type level `And` function.